### PR TITLE
[14.0][MIG] hs_code_id company_dependent=True to False

### PR DESCRIPTION
**Pull request by alexis-via**
Migration script provided for product.template and product.category
Fixes bug #126

### CODE REVIEW

**Comment by alexis-via** [comment]
Note: we have to merge this BEFORE migration to v15 starts #144

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
I think you should do an SQL query on this form instead of such complicated code that is going to be very slow:

```sql
UPDATE product_category pc
SET hs_code_id = SUBSTRING(ip.value_reference, 7, 99)::int
FROM ir_propery ip
WHERE ip.res_id = pc.id AND ip.name = 'hs_code_id'
```
The number 7 is the position for starting to extract the number, and I'm not sure if it's the right one, but I think you get the point.

**Comment by alexis-via** [comment]
SQL 1, ORM 0

File: product_harmonized_system/migrations/14.0.2.0.0/post-migration.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        ).id
+        cr.execute(
+            """
+            UPDATE product_category pc
+            SET hs_code_id=SUBSTRING(ip.value_reference, 9, 99)::int
+            FROM ir_property ip
+            WHERE ip.res_id like 'product.category,%%' AND
+            SUBSTRING(ip.res_id, 18, 99)::int=pc.id AND
+            ip.name='hs_code_id' AND
+            ip.company_id IS NOT null AND
```
Comment (pedrobaeza) [inline product_harmonized_system/migrations/14.0.2.0.0/post-migration.py]: I don't think you should restrict this. Maybe there's a generic value for all companies in the DB. ir.property allows it...

File: product_harmonized_system/migrations/14.0.2.0.0/post-migration.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            ip.value_reference IS NOT null AND
+            ip.type='many2one' AND
```
Comment (pedrobaeza) [inline product_harmonized_system/migrations/14.0.2.0.0/post-migration.py]: If you are already matching by the field id, I think this is too much, but I don't think this is going to increase too much the query time.

**Comment by pedrobaeza** [comment]
> SQL 1, ORM 0

Welcome to the usual way on OpenUpgrade!

File: product_harmonized_system/migrations/14.0.2.0.0/post-migration.py (hunk 1 above)
Comment (alexis-via) [reply product_harmonized_system/migrations/14.0.2.0.0/post-migration.py]: done

File: product_harmonized_system/migrations/14.0.2.0.0/post-migration.py (hunk 2 above)
Comment (alexis-via) [reply product_harmonized_system/migrations/14.0.2.0.0/post-migration.py]: done

**Review by pedrobaeza — APPROVED** [review]

**Review by sebalix — APPROVED** [review]

File: product_harmonized_system/migrations/14.0.2.0.0/post-migration.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        pt_field_id = env.ref(
+            "product_harmonized_system.field_product_template__hs_code_id"
+        ).id
+        cr.execute(
+            """
+            UPDATE product_template pt
+            SET hs_code_id=SUBSTRING(ip.value_reference, 9, 99)::int
+            FROM ir_property ip
+            WHERE ip.res_id like 'product.template,%%' AND
+            SUBSTRING(ip.res_id, 18, 99)::int=pt.id AND
```
Comment (hparfr) [inline product_harmonized_system/migrations/14.0.2.0.0/post-migration.py:39]: Shound't be 17 instead ?

`SUBSTRING(ip.res_id, LENGTH('product.product,') +1 )::int=pt.id AND`

Comment (hparfr) [reply product_harmonized_system/migrations/14.0.2.0.0/post-migration.py:39]: my bad, was reviewing the wrong field !

**Pull request merged**